### ISSUE TITLE
Protect the routes!!!

### ISSUE THREAD

**Issue by Kiranism** [webdevcody/wdc-saas-starter-kit#12]
@webdevcody Hey, what can we do to protect the routes? Since we aren't using middleware, maybe we can do the checks directly in the pages?

something like
`export default async function Dashboard() {
const user = await getUser();
  if (!user) {
    return redirect('/admin/login');
  }
  return (...)`

![image](https://github.com/user-attachments/assets/d88e8476-3111-4039-bf4b-339e6352c37d)

**Comment by Kiranism** [comment]
Yeah, that will work. Maybe that can be added to the code.

**Comment by evert-arias** [comment]
It's already in the code. The dashboard page uses the `assertAuthenticated` function which is implemented in lib/session.ts.

**Comment by Kiranism** [comment]
> It's already in the code. The dashboard page uses the `assertAuthenticated` function which is implemented in lib/session.ts.

It just throwing an error. Please check this picture https://github.com/webdevcody/wdc-saas-starter-kit/issues/12#issue-2432150213

**Comment by evert-arias** [comment]
@Kiranism I see. There were a few issues I had to solve to get this template work on production. I'll check that out.

**Comment by evert-arias** [comment]
So, when built for production, `isAuthenticationError` evaluates as false, but not when running locally in dev mode. As a result, it is showing the generic error message instead of the authentication error message.

<img width="793" alt="image" src="https://github.com/user-attachments/assets/cf654dfc-fad6-479c-b8f9-8fb7bdd37575">

**Comment by webdevcody** [comment]
I think having a method called assertAuthenticatedAndRedirect('/your/url') would be my recommendation.  I personally don't like when a user is just randomly redirected to a new page; I'd rather show them an error so they know not to do it again.  I can look into this error difference between prod and dev and try to update this.

**Comment by evert-arias** [comment]
I concur with that strategy.

**Comment by Kiranism** [comment]
> I think having a method called assertAuthenticatedAndRedirect('/your/url') would be my recommendation. I personally don't like when a user is just randomly redirected to a new page; I'd rather show them an error so they know not to do it again. I can look into this error difference between prod and dev and try to update this.

Fair enough. I also noticed that you can still visit the login page (auth routes) after signing in, which can be avoided with a similar check."
